CODE_OF_CONDUCT.md

```diff
@@ -1,56 +1,154 @@
+may take action they deem appropriate, up to and including temporary ban or permanent
+expulsion from the community _without warning_ (and without refund in the case of
+a paid event). If you have been involved in unacceptable behavior with current Habitat
+community members outside the boundaries of the Habitat Community, the Community Organizers 
+retain the right to treat those external incidents in the same manner as internal incidents.
+
+Any physical violence _or_ intimidation, threatened or acted on, is a serious
+offense and will result in immediate exclusion from the community and appropriate
+follow up with law enforcement (no, we're not kidding).
+
+## Procedure for Handling Disagreements and Incidents
+Disagreements are inherent to a group of impassioned people. When they occur, we seek to resolve disagreements and differing views constructively and with the help of the community and community processes. When disagreements escalate, we ask our community advocates to step in to moderate, mediate, and help resolve tense situations.
+
+The Habitat Community advocates are well informed on how to deal with incidents. Report the incident (preferably in writing) to one of the community advocates listed above. See the "Roles" section for details on each role.
+
+### Handling Incidents
+
+When a Community Organizer or Project Maintainer notices someone behaving in a way that is outside of our guidelines (a violator) the Advocate should make every reasonable attempt to help curtail that behavior.  The Advocate may:
+
+* Remind the violator about our Community Code of Conduct and provide a link to this document.
+* Ask the violator to stop the unacceptable behavior.
+* Raise the issue with a maintainer, the community manager, or any member of the core project team.
+* Allow time for the violator to correct the behavior.
+
+The Advocate should take the following steps if the behavior is not brought inline with our guidelines or the incident is not resolved.
+
+* Consult with another Community Organizer to make a judgement call about what reasonable corrective actions are warranted.
+* Apply the corrective action.
+* Document the incident.
+
+#### Documenting Incidents
+All incident reports will be kept in a private repository that is shared with the aforementioned Community Advocates, Community Manager and Decider under the *Roles* section. No other individuals or project contributors will be given access to these incident reports. On the displacement of
+any Community Organizer in the *Roles* list above, that individual will immediately lose access
+to this repository.
+
+The important information to report consists of:
+
+* Identifying information (name, email address, Slack nick, etc.) of the person doing the harassing
+* The behavior that was in violation
+* The approximate time of the behavior
+* The circumstances surrounding the incident
+* Other people involved in the incident
```

Comment:
My initial feeling is to say that we should track an initial point of contact for the victim. Do you have any opinions on that?